sensor/extrinsic_reflector_based_calibrator/src/extrinsic_reflector_based_calibrator.cpp

```diff
@@ -1280,21 +1336,169 @@ void ExtrinsicReflectorBasedCalibrator::calibrateSensors()
       "The calibrated poses differ considerably with the initial calibration. This may be either a "
       "fault of the algorithm or a bad calibration initialization");
   }
+
+  output_metrics_.push_back(static_cast<float>(converged_tracks_.size()));
+  output_metrics_.push_back(static_cast<float>(calibrated_2d_distance_error));
+  output_metrics_.push_back(static_cast<float>(calibrated_2d_yaw_error));
+}
+
+void ExtrinsicReflectorBasedCalibrator::findCombinations(
+  int n, int k, std::vector<int> & curr, int first_num,
+  std::vector<std::vector<int>> & combinations)
+{
+  int curr_size = static_cast<int>(curr.size());
+  if (curr_size == k) {
+    combinations.push_back(curr);
+    return;
+  }
+
+  int need = k - curr_size;
+  int remain = n - first_num + 1;
+  int available = remain - need;
+
+  for (int num = first_num; num <= first_num + available; num++) {
+    curr.push_back(num);
+    findCombinations(n, k, curr, num + 1, combinations);
+    curr.pop_back();
+  }
+
+  return;
+}
+
+void ExtrinsicReflectorBasedCalibrator::crossValEvaluation(
+  pcl::PointCloud<PointType>::Ptr lidar_points_pcs,
+  pcl::PointCloud<PointType>::Ptr radar_points_rcs)
+{
+  int tracks_size = static_cast<int>(converged_tracks_.size());
+  if (tracks_size <= 3) return;
+
+  pcl::PointCloud<PointType>::Ptr crossval_lidar_points_pcs(new pcl::PointCloud<PointType>);
+  pcl::PointCloud<PointType>::Ptr crossval_radar_points_rcs(new pcl::PointCloud<PointType>);
+  pcl::registration::TransformationEstimationSVD<PointType, PointType> crossval_estimator;
+  Eigen::Matrix4f crossval_radar_to_parent_transformation;
+  Eigen::Isometry3d crossval_calibrated_2d_radar_to_parent_transformation;
+  Eigen::Isometry3d crossval_calibrated_2d_radar_to_lidar_transformation;
+
+  double total_crossval_calibrated_2d_distance_error;
```

Comment:
This is just my opinion, but it is better to user spacial locality when declaring variables.
Outside the for loop the variables are not used, so it is safer to declare it inside
(the compiler will probably still declare it outside but that is another story)